packages/pds/src/repo/prepare.ts

```diff
@@ -41,29 +45,35 @@ export const blobsForWrite = (record: any): PreparedBlobRef[] => {
       })
     }
     return refs
-  } else if (record.$type === lex.ids.AppBskyFeedPost) {
+  } else if (isPost(record)) {
     const refs: PreparedBlobRef[] = []
-    const embed = record?.embed
-    if (embed?.$type === 'app.bsky.embed.images') {
-      const doc = lex.schemaDict.AppBskyEmbedImages
-      for (let i = 0; i < embed.images?.length || 0; i++) {
-        const img = embed.images[i]
+    const embeds = isComplexRecordEmbed(record.embed)
+      ? [
+          { $type: 'app.bsky.embed.record', ...record.embed.record },
+          record.embed.media,
+        ]
+      : record.embed
```

Comment:
same here

Reply:
Extracted this into a function for readability in both places 👍